rb/cluster.py

```diff
@@ -290,3 +293,100 @@ def all(self, timeout=None, max_concurrency=64, auto_batch=True):
         return self.fanout('all', timeout=timeout,
                            max_concurrency=max_concurrency,
                            auto_batch=auto_batch)
+
+    def execute_commands(self, mapping, *args, **kwargs):
+        """Concurrently executes a sequence of commands on a Redis cluster that
+        are associated with a routing key, returning a new mapping where
+        values are a list of results that correspond to the command in the same
+        position. For example::
+
+            >>> cluster.execute_commands({
+            ...   'foo': [
+            ...     ('PING',),
+            ...     ('TIME',),
+            ...   ],
+            ...   'bar': [
+            ...     ('CLIENT', 'GETNAME'),
+            ...   ],
+            ... })
+            {'bar': [<Promise None>],
+             'foo': [<Promise True>, <Promise (1454446079, 418404)>]}
+
+        Commands that are instances of :class:`redis.client.Script` will first
+        be checked for their existence on the target nodes then loaded on the
+        targets before executing and can be interleaved with other commands::
+
+            >>> from redis.client import Script
+            >>> TestScript = Script(None, 'return {KEYS, ARGV}')
+            >>> cluster.execute_commands({
+            ...   'foo': [
+            ...     (TestScript, ('key:1', 'key:2'), range(0, 3)),
+            ...   ],
+            ...   'bar': [
+            ...     (TestScript, ('key:3', 'key:4'), range(3, 6)),
+            ...   ],
+            ... })
+            {'bar': [<Promise [['key:3', 'key:4'], ['3', '4', '5']]>],
+             'foo': [<Promise [['key:1', 'key:2'], ['0', '1', '2']]>]}
+
+        Internally, :class:`FanoutClient` is used for issuing commands.
+        """
+        def is_script_command(command):
+            return isinstance(command[0], Script)
+
+        def check_script_load_result(script, result):
+            if script.sha != result:
+                raise AssertionError(
+                    'Hash mismatch loading {!r}: expected {!r}, got {!r}'.format(
+                        script,
+                        script.sha,
+                        result,
+                    )
+                )
```

Comment:
I changed this to an exception, as it'd throw `NoScriptError` on the next command anyway.